packages/esm-bed-management-app/src/bed-administration/form/add-edit-bed.workspace.tsx

```diff
@@ -0,0 +1,407 @@
+import React, { useEffect } from 'react';
+import {
+  type DefaultWorkspaceProps,
+  ResponsiveWrapper,
+  useLayoutType,
+  useSession,
+  showSnackbar,
+} from '@openmrs/esm-framework';
+import { useTranslation } from 'react-i18next';
+import { Controller, useForm } from 'react-hook-form';
+import styles from './add-edit-bed.workspace.scss';
+import {
+  ButtonSet,
+  Button,
+  InlineLoading,
+  TextInput,
+  NumberInput,
+  TextArea,
+  Select,
+  SelectItem,
+  ComboBox,
+  Form,
+  MultiSelect,
+  Stack,
+  Tag,
+} from '@carbon/react';
+import classNames from 'classnames';
+import { z } from 'zod';
+import { zodResolver } from '@hookform/resolvers/zod';
+import capitalize from 'lodash-es/capitalize';
+import { useBedTags, useLocationsWithAdmissionTag } from '../../summary/summary.resource';
+import { type InitialData } from '../../types';
+import { editBed, saveBed, useBedType } from './add-edit-bed.resource';
+
+type AddEditBedWorkspaceProps = DefaultWorkspaceProps & {
+  bed?: InitialData;
+  mutateBeds: () => void;
+  defaultLocation?: { display: string; uuid: string };
+};
+
+const AddEditBedWorkspace: React.FC<AddEditBedWorkspaceProps> = ({
+  closeWorkspace,
+  closeWorkspaceWithSavedChanges,
+  promptBeforeClosing,
+  bed,
+  mutateBeds,
+  defaultLocation,
+}) => {
+  const { t } = useTranslation();
+  const session = useSession();
+  const locationUuid = session?.sessionLocation?.uuid;
+  const isTablet = useLayoutType() === 'tablet';
+
+  const { admissionLocations } = useLocationsWithAdmissionTag();
+  const { bedTypes } = useBedType();
+  const { bedTags } = useBedTags();
+
+  const numberInString = z.string().transform((val, ctx) => {
+    const parsed = parseInt(val);
+    if (isNaN(parsed) || parsed < 1) {
+      ctx.addIssue({
+        code: z.ZodIssueCode.custom,
+        message: t('validNumberRequired', 'Please enter a valid number'),
+      });
+      return z.NEVER;
+    }
+    return val;
+  });
+
+  const bedFormSchema = z.object({
+    bedId: z
+      .string()
+      .min(5, t('bedIdMinLength', 'Bed ID must be at least 5 characters'))
+      .max(255, t('bedIdMaxLength', 'Bed ID must not exceed 255 characters')),
```

Comment:
```suggestion
      .min(1, t('bedNumberRequired', 'Bed number is required'))
      .max(10, t('bedNumberMaxLength', 'Bed number must not exceed 10 characters')),
```

Doesn't this differ from the [backend constraint](https://github.com/openmrs/openmrs-module-bedmanagement/blob/master/api/src/main/resources/Bed.hbm.xml#L35) in the hibernate mapping? Here's a snippet from the error message you get when you try to create a bed with the bed number greater than 10:

```
"error": {
  "message": "['Bed[hashCode=f51e6c51,uuid=0a60213d-8e0e-49f3-9b0c-34e341cf058f]' failed to validate with reason: bedNumber: This value exceeds the maximum length of 10 permitted for this field.]",
  "code": "org.openmrs.validator.ValidateUtil:88",
}
````